src/box/lua/trigger.c

```diff
@@ -0,0 +1,242 @@
+	event_trigger_iterator_create(&it, event);
+	struct func_adapter *trg = NULL;
+	const char *name = NULL;
+	int idx = 0;
+	lua_createtable(L, 0, 0);
+	while (event_trigger_iterator_next(&it, &trg, &name)) {
+		idx++;
+		lua_createtable(L, 2, 0);
+		lua_pushstring(L, name);
+		lua_rawseti(L, -2, 1);
+		func_adapter_lua_get_func(trg, L);
+		lua_rawseti(L, -2, 2);
+		lua_rawseti(L, -2, idx);
+	}
+	lua_setfield(L, -2, event->name);
+	return true;
+}
+
+/**
+ * Pushes a key-value table, where the key is the event name and value is an
+ * array of triggers, represented by two-element [trigger_name, trigger_handler]
+ * arrays, registered on this event, in the order in which they will be called.
+ * If an event name is passed, a table contains only one key which is passed
+ * argument, if there is an event with such a name, or returned table is empty,
+ * if the event does not exist.
+ */
+static int
+luaT_trigger_info(struct lua_State *L)
+{
+	if (lua_gettop(L) == 0) {
+		lua_createtable(L, 0, 0);
+		bool ok = event_foreach(trigger_info_push_event, L);
+		assert(ok);
+		(void)ok;
+	} else {
+		const char *event_name = luaL_checkstring(L, 1);
+		struct event *event = event_get(event_name, false);
+		if (event == NULL || !event_has_triggers(event)) {
+			lua_createtable(L, 0, 0);
+			return 1;
+		}
+		lua_createtable(L, 0, 1);
+		trigger_info_push_event(event, L);
+	}
+	return 1;
+}
+
+const char *event_trigger_iterator_typename = "trigger.event_trigger_iterator";
+
+/**
+ * Lua-iterator over triggers of the event.
+ */
+struct lua_event_trigger_iterator {
+	/** An iterator over triggers of the event. */
+	struct event_trigger_iterator it;
+};
+
+/**
+ * Gets lua_event_trigger_iterator from Lua stack with type check.
+ */
+static inline struct lua_event_trigger_iterator *
+lua_check_event_trigger_iterator(struct lua_State *L, int idx)
+{
+	return luaL_checkudata(L, idx, event_trigger_iterator_typename);
+}
+
+/**
+ * Takes an iterator step.
+ */
+static int
+luaT_trigger_iterator_next(struct lua_State *L)
+{
+	struct lua_event_trigger_iterator *it =
+		lua_check_event_trigger_iterator(L, 1);
+	struct func_adapter *trigger = NULL;
+	const char *name = NULL;
+	if (event_trigger_iterator_next(&it->it, &trigger, &name)) {
+		lua_pushstring(L, name);
+		func_adapter_lua_get_func(trigger, L);
+		return 2;
+	}
+	return 0;
+}
+
+/**
+ * Destroys an iterator.
+ */
+static int
+luaT_trigger_iterator_gc(struct lua_State *L)
+{
+	struct lua_event_trigger_iterator *it =
+		lua_check_event_trigger_iterator(L, 1);
+	event_trigger_iterator_destroy(&it->it);
+	TRASH(it);
+	return 0;
+}
+
+/**
+ * Creates iterator over triggers of event with passed name.
+ * The iterator yields a pair [trigger_name, trigger_handler].
+ * Return next method of iterator and iterator itself.
+ */
+static int
+luaT_trigger_pairs(struct lua_State *L)
+{
+	const char *event_name = luaL_checkstring(L, 1);
+	struct event *event = event_get(event_name, false);
+	if (event == NULL)
+		return 0;
```

Comment:
According to the coveralls report, this line isn't covered by any tests. Please add a test for this case.

Reply:
Added the check of pairs on empty event.